suite/suite.go

```diff
@@ -0,0 +1,256 @@
+//
+// In general, using SuiteManager will help make sure your fixture files and test cases
+// stay aligned as the code evolves: if you remove tests that used some hunks, you'll get notified;
+// if you add tests that expect certain patterns of hunks in some files, but didn't write them yet,
+// you'll get notified; etc.
+//
+// Approximately the only thing SuiteManager can't help you with is if you use whole files for organization,
+// and you remove some files (or add some features or tests that would expect whole new files),
+// and you're using globbing to register filenames, then SuiteManager can't help you infer what's missing.
+type SuiteManager struct {
+	fs fsx.FS
+
+	workset map[string]fileContentExpectations
+}
+
+type fileContentExpectations struct {
+	filename string
+
+	handlers map[HunkGlob]TestingPattern
+}
+
+// TestingPattern is the interface that test code should implement to become
+// usable by SuiteManager as a handler.
+//
+// Examples of TestingPattern that are well-known and easy to use immediately
+// include `testexec.TestingPattern`
+//
+// When implementing this interface:
+// The Run method will do the majority of the work.
+// Other methods on the interface let the SuiteManager know how recursion should work,
+// and provide some information that may be used in logging.
+//
+// Note that it is not necessary to import this package in order to implement this interface.
+// Only symbols from testmark's core package and the go standard library are used.
+type TestingPattern interface {
+	Run(
+		t *testing.T, // The standard testing object, for obvious purposes.
+		filename string, // The filename the subject data was loaded from.  Typically not needed (data comes in via `subject` and, if appropriate, can flow out through `patchAccum`).
+		subject *testmark.DirEnt, // The subject hunk (and enclosing dirent, in case you want to navigate to child hunks).
+		reportUse func(string), // Should be called with the full path of any hunk that's consumed by this test.  Used to detect orphaned hunks that went unused by the whole suite.
+		reportUnrecog func(string), // If this test code owns all child hunks, it may call this to report one that it doesn't recognize.
+		patchAccum *testmark.PatchAccumulator, // If non-nil, means regenerating golden master data is requested instead of testing.
+	) error // Run may return errors or call t.Fatal itself.
+
+	Name() string // Purely for diagnostic purposes.
+
+	OwnsAllChildren() bool // If true, the suite manager will not look at any hunks beneath this subject or attempt to match other globs and testing patterns to them.
+}
+
+type FilenameGlob string
+type HunkGlob string
+
+// WorkWith registers one or more files to be handled with some TestingPattern.
+// A globbing pattern can be used for both the filenames, and for what hunk labels
+// within the file should be handled.
+//
+// WorkWith can be called more than one time with the name filename, or with filename
```

Comment:
```suggestion
// WorkWith can be called more than one time with the same filename, or with filename
```
Pretty sure you meant this?